src/docfx/build/link/LinkResolver.cs

```diff
@@ -147,6 +151,11 @@ internal class LinkResolver
 
             if (error is null && _buildScope.OutOfScope(file))
             {
+                if (file.Origin == FileOrigin.Dependency && _buildScope.GetContentType(file) == ContentType.Resource)
```

Comment:
how about updating [BuildScope.cs](https://github.com/dotnet/docfx/blob/a6583ad73bb75f83b4c252a4bcd42c21685d3c60/src/docfx/build/context/BuildScope.cs#L182) instead? Include the dependency resources to the build scope, the pipeline should just work.

Reply:
As discussed, that way all CRR resource will be built and published

Reply:
Offline synced with @mingwli , the dynamic scope of dependency resources also breaks the design presumption of `PublishUrlMap`, which works on static build scope.
Consider this case, if the user defines a redirect URL of existed resource file from dependency, docfx would not be able to detect the output path conflict.

Reply:
NOT all CRR resource will be published if the user config the right scope.